### PR TITLE
Add outputOptionsPrintStringLitsLiterally

### LINKED ISSUE #64: pPrint prints unexpected characters for unprintable string contents
For example

```haskell
pPrint ("\DC1\205N\237\232s\225\232N\147K\173\RSE\201\EM" :: ByteString)
"ÍNíèsáèNK­EÉ"
```

I would expect `pPrint (x :: Bytestring)` to be `x`

### PR DESCRIPTION
**Pull request by expipiplus1**
Closes #64

### CODE REVIEW

**Comment by expipiplus1** [comment]
It would be very nice to turn this on by default, but then one bumps into https://gitlab.haskell.org/ghc/ghc/issues/11529

File: src/Text/Pretty/Simple/Internal/OutputPrinter.hs [hunk 1]
```diff
@@ -74,6 +74,16 @@ data OutputOptions = OutputOptions
   , outputOptionsEscapeNonPrintable :: Bool
   -- ^ Whether to replace non-printable characters with hexadecimal escape
   -- sequences.
+  , outputOptionsPrintStringLitsLiterally :: Bool
```
Comment (cdepillabout) [inline src/Text/Pretty/Simple/Internal/OutputPrinter.hs:77]: Would it make sense to turn these two options into a data type with three constructors?

That might follow the flow of the code better below in the nested `if` statements.

File: src/Text/Pretty/Simple.hs [hunk 2]
```diff
@@ -665,3 +665,12 @@ pStringOpt outputOptions =
...
 -- "this string has non-printable characters: \x8 and \x9"
+--
+-- If outputOptionsPrintStringLitsLiterally is set strings are
+-- replicated faithfully.
+--
+-- >>> pPrintStringOpt CheckColorTty defaultOutputOptionsDarkBg{ outputOptionsPrintStringLitsLiterally = False } "\"Several encodings of A: A \\65 \\065 \\x41 \\n\""
+-- "Several encodings of A: A A A A "
+--
+-- >>> pPrintStringOpt CheckColorTty defaultOutputOptionsDarkBg{ outputOptionsPrintStringLitsLiterally = True } "\"Several encodings of A: A \\65 \\065 \\x41 \\n\""
+-- "Several encodings of A: A \65 \065 \x41 \n"
```
Comment (cdepillabout) [inline src/Text/Pretty/Simple.hs:676]: This is pretty neat!  I bet there are definitely users that wanted something like this.

Comment (expipiplus1) [reply src/Text/Pretty/Simple.hs:676]: Thanks :)

File: src/Text/Pretty/Simple/Internal/OutputPrinter.hs (hunk 1 above)
Comment (expipiplus1) [reply src/Text/Pretty/Simple/Internal/OutputPrinter.hs:77]: Good idea

Comment (expipiplus1) [reply src/Text/Pretty/Simple/Internal/OutputPrinter.hs]: Done, feel free to suggest some better names.

**Comment by cdepillabout** [comment]
Rebased on top of `master` in #69.

**Pull request closed without merging**